pkgs/development/compilers/temurin/default.nix

```diff
@@ -0,0 +1,445 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  removeReferencesTo,
+  autoconf,
+  bash,
+  which,
+  zip,
+  unzip,
+  pkg-config,
+  cpio,
+  file,
+  openjdk11,
+  openjdk17,
+  openjdk21,
+  openjdk23,
+  openjdk24,
+  cups,
+  freetype,
+  alsa-lib,
+  libjpeg,
+  giflib,
+  libpng,
+  zlib,
+  lcms2,
+  fontconfig,
+  xorg,
+  cctools,
+  darwin,
+}:
+
+let
+  mkTemurinJDK =
+    {
+      featureVersion,
+      version,
+      build,
+      hash,
+    }:
+    let
+      bootstrapJdk =
+        if featureVersion == "11" then
+          openjdk11
+        else if lib.versionAtLeast featureVersion "24" then
+          openjdk24
+        else if lib.versionAtLeast featureVersion "23" then
+          openjdk23
+        else if lib.versionAtLeast featureVersion "21" then
+          openjdk21
+        else
+          openjdk17;
+
+      buildConfigMap = {
+        x86_64-linux =
+          if featureVersion == "11" then
+            "linux-x86_64-normal-server-release"
+          else
+            "linux-x86_64-server-release";
+        aarch64-linux =
+          if featureVersion == "11" then
+            "linux-aarch64-normal-server-release"
+          else
+            "linux-aarch64-server-release";
+        x86_64-darwin =
+          if featureVersion == "11" then
+            "macosx-x86_64-normal-server-release"
+          else
+            "macosx-x86_64-server-release";
+        aarch64-darwin =
+          if featureVersion == "11" then
+            "macosx-aarch64-normal-server-release"
+          else
+            "macosx-aarch64-server-release";
+      };
```

Comment:
How about:

```nix
let
  supportedSystems = [
    "x86_64-linux"
    "aarch64-linux"
    "x86_64-darwin"
    "aarch64-darwin"
  ];
in
buildConfigMap = lib.genAttrs supportedSystems (system:
  let
    parts = lib.strings.splitString "-" system;
    arch = lib.elemAt parts 0;
    os = lib.elemAt parts 1;

    osPrefix = {
      linux = "linux";
      darwin = "macosx";
    }.${os};

    normalPart = if featureVersion == "11" then "-normal" else "";
  in
    "${osPrefix}-${arch}${normalPart}-server-release"
);
```

Reply:
But i prefer explicit version.
Because is also more readable.

Reply:
Yes it is explicit... but there are plenty of duplicated things. Don't you think we should strive for conciseness and avoid duplicating conditions and strings?

Reply:
I think is best to DRY

Reply:
I also think that. Thank you!